native/native_main.nim

```diff
@@ -0,0 +1,89 @@
+# compile with 
+# nim c --verbosity:0 native_main.nim
+
+# test with
+# time printf '%c\0\0\0{"cmd": "run", "command": "echo $PATH"}' 39 | ./native_main
+
+# compare with
+# time printf '%c\0\0\0{"cmd": "run", "command": "echo $PATH"}' 39 | python ~/.local/share/tridactyl/native_main.py
+
+import json
+import options
+import osproc
+# https://nim-lang.org/docs/json.html
+
+let VERSION = "0.2.0"
+
+type 
+    MessageResp* = object
+        cmd*, version*, content*, error*, command*: Option[string]
+        code: Option[int]
+
+type 
+    MessageRecv* = object
+        cmd*, version*, content*, error*, command*: Option[string]
+        code: Option[int]
+
+# let a = MessageResp(cmd: some(""))
+# echo(a)
+
+proc getMessage(): MessageRecv =
+
+    # length of the string - not required AFAICT
+    var length: int32
+    discard readBuffer(stdin, addr(length), 4)
+
+    return to(parseJson(readAll(stdin)),MessageRecv)
+
+proc handleMessage(msg: MessageRecv): MessageResp =
+
+    let cmd = msg.cmd.get()
+    var command: MessageResp
+
+    case cmd:
+        of "run":
+            command.content = some(execProcess(msg.command.get(), options={poEvalCommand,poStdErrToStdOut}))
+            command.code = some(0) # need to use something other than execProcess to capture this
+
+        of "version":
+            command.version = some(VERSION)
+
+    return command
+
+write(stdout, %* handleMessage(getMessage())) # %* converts the object to JSON
```

Comment:
This apparently isn't sufficient for Firefox to understand it. Slightly perplexed as in my terminal the output appears similar, albeit with e.g. `"cmd": null` for all the missing values in Nim. Further investigation needed.